services/editor/src/contexts/CurrentUser.js

```diff
@@ -1,38 +1,27 @@
-import React, { Component } from 'react';
+import React, { useContext, useEffect, useState } from 'react';
 import { tweekManagementClient } from '../utils/tweekClients';
 
-export const CurrentUserContext = React.createContext(null);
+const CurrentUserContext = React.createContext(null);
+CurrentUserContext.displayName = 'CurrentUser';
 
-export class CurrentUserProvider extends Component {
-  state = {
-    currentUser: null,
-  };
-  _isMounted = false;
+export const CurrentUserProvider = ({ children }) => {
+  const [currentUser, setCurrentUser] = useState(null);
+  useEffect(() => {
+    let cancel = false;
 
-  async componentDidMount() {
-    this._isMounted = true;
-    const currentUser = await tweekManagementClient.currentUser();
-    if (!this._isMounted) {
-      return;
+    async function run() {
+      const user = await tweekManagementClient.currentUser();
+      if (!cancel) {
```

Comment:
Do we need this safeguard?

Reply:
it's kinda like check `isMounted` in lifecycle methods, there's no harm in it